app/controllers/course/video/submission/submissions_controller.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 class Course::Video::Submission::SubmissionsController < Course::Video::Submission::Controller
   before_action :authorize_video!, only: :create
+  skip_authorize_resource :submission, only: :edit
```

Comment:
If you remove this declaration and allow the implicit `load_and_authorize_resource` to run, can you still `rescue CanCan::AccessDenied`? Any reason for making this explicit?

Reply:
Can you try `rescue_from` - not sure whether it will work?

Reply:
Nope, it still ends up on access denied. I think the issue here is that the `load_and_authorize_resource` is declared in the superclass. So when it fails raises in the superclass itself, and the method in this current class is never called (I did a `byebug` within the rescue block and it was never triggered).

Reply:
Ok then. Can I bother you to explain this a little in the `edit` method, just so we understand?

Reply:
Sure, I'll add a comment.